### PR TITLE
[MM-34338] Migrate 'components/post_profile_picture' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by arvinDarmawan**
Summary
This code migrate "post_profile_picture" component to TypeScript. I add the example test case for "post_profile_picture" component.

Ticket Link
Jira card: https://mattermost.atlassian.net/browse/MM-34338
Github Issue: https://github.com/mattermost/mattermost-server/issues/17265

```release-note
Release Note
- Refactor component to typescript, no changes in UI
```

### CODE REVIEW

File: components/post_profile_picture/post_profile_picture.tsx [hunk 1]
```diff
@@ -112,12 +114,11 @@ export default class PostProfilePicture extends React.PureComponent {
                 profileSrc={profileSrc}
                 isEmoji={isEmoji}
                 status={status}
-                userId={user ? user.id : null}
+                userId={user ? user.id : undefined}
```
Comment (lieut-data) [inline components/post_profile_picture/post_profile_picture.tsx:117]: Could we use `user.?id` here instead?

File: components/post_profile_picture/post_profile_picture.tsx [hunk 2]
```diff
@@ -112,12 +114,11 @@ export default class PostProfilePicture extends React.PureComponent {
...
                 channelId={post.channel_id}
-                username={user ? user.username : null}
+                username={user ? user.username : undefined}
```
Comment (lieut-data) [inline components/post_profile_picture/post_profile_picture.tsx:119]: Could we use `user.?username` here instead?

**Review by lieut-data — APPROVED** [review]
Thanks, @arvinDarmawan!

**Review by devinbinnie — APPROVED** [review]
LGTM other than @lieut-data's comments. Thanks @arvinDarmawan!

File: components/post_profile_picture/post_profile_picture.tsx (hunk 1 above)
Comment (arvinDarmawan) [reply components/post_profile_picture/post_profile_picture.tsx]: Hi, thank you for the feedback. I already change it to  `user?.id`

File: components/post_profile_picture/post_profile_picture.tsx (hunk 2 above)
Comment (arvinDarmawan) [reply components/post_profile_picture/post_profile_picture.tsx]: Hi, I already change it to  `user?.username`. Thank you.

**Comment by arvinDarmawan** [comment]
Hi, wanna know if this code already begin to be review   @ogi-m  ? Thank you.

**Review by ogi-m — APPROVED** [review]
Thanks @arvinDarmawan and sorry for the wait! LGTM

**Pull request merged**